python/cudf/cudf/core/column/string.py

```diff
@@ -1786,6 +1791,66 @@ def translate(self, table, **kwargs):
             cpp_translate(self._column, table), **kwargs
         )
 
+    def normalize_spaces(self):
+        return cpp_normalize_spaces(self._column)
```

Comment:
Needs docstring. Also This will return a `Column` instead of  `Series / Index`. Could you use the `_return_or_inplace` function to handle constructing based on the parent or something similar? Ideally we should maintain at least the name of the input, and in the case of the Series if the length of the Column is the same as the input we should maintain the index.